src/reconfigurability.jl

```diff
@@ -83,7 +83,7 @@ function empirical_gramian(f::Function, g::Function, M::Int, N::Int, L::Int; opt
                         pnd = pr[:, k] + en[N+1:end]
                         y = ssp2(f, g, dt, tf, xnd, up, pnd)
                         y = y ./ xm[a, d]
-                        o[:, a] = y'
+                        o[:, a] = vec(y)
```

Comment:
```suggestion
                        o[:, a] = y
```
제가 코드를 제대로 이해 못 했을수도 있는데, `ssp2` 아웃풋인 `y` 가 애초에 벡터로 보입니다.
그럼 그냥 `y` 로 넣어도 될 것 같은데, 맞을까요?

Reply:
애초에 `y0`는 output 벡터를 나타내고, `y`는 시간에 따른 trajectory로 행렬이 됩니다.

Reply:
아직 잘 이해가 안 가는데...
데이터의 차원을 설명해주시겠어요?

`y0` 는 왜 갑자기 설명이 나오는거죠? 제가 따라가질 못하는 것 같은데...

Reply:
음 그러니까, output의 차원을 l이라고 할 때, `y0`는 `lx1` 크기의 행렬을 나타냅니다.
그리고 `y`는 `y0`를 초기값으로 하여 time span에 대한 궤적을 나타내는 값으로, `lxlength(tspan)` 크기의 행렬이 됩니다.

Reply:
아 124번 라인에 `y0` 가 있구나... 이걸 의미하는건가요?

그 부분은 좀 이해된 것 같습니다.

그럼 line 89 는 어떻게 얻어지는건지, 논문의 수식 번호 등으로 알려주실 수 있을까요?

코드상으로 헷갈리는게, `ssp2` 의 아웃풋인 `y` 가 trajectory 인지, line 85의 덮어쓴 `y` 가 trajectory 인지도 헷갈리네요...


저는 참고로 아래 논문을 보고 있는데, 이 논문이 맞죠?
`K`, `D`, `A` 같은건 안 보여서... 혹시 다른 논문인가요?
[1] M. Tahavori and A. Hasan, “Fault Recoverability for Nonlinear Systems with Application to Fault Tolerant Control of UAVs,” Aerospace Science and Technology, vol. 107, p. 106282, Dec. 2020, doi: [10.1016/j.ast.2020.106282](https://doi.org/10.1016/j.ast.2020.106282).

Reply:
네 `y0`는 그걸 의미하는게 맞습니다. 논문의 수식번호로 표기하기에 어려운것이 [1]의 논문을 보면 empirical gramian을 구하는 식이 엄청나게 간단하게만 소개되어 있습니다ㅠ 그래서 이런저런 조사를 했고, 다음 사이트를 함께 참고하여 코드를 작성했습니다. (이것은 코드에도 명시되어있습니다.)
[2] https://gramian.de/
[3] https://github.com/gramian/emgr

Reply:
아하...
해당 사이트들은 논문과 부합하는건가요? 아니면 수정된 부분이 있나요?

그리고 이 부분에 대한 설명도 부탁드립니다.


> 코드상으로 헷갈리는게, `ssp2` 의 아웃풋인 `y` 가 trajectory 인지, line 85의 덮어쓴 `y` 가 trajectory 인지도 헷갈리네요...

Reply:
음... 아니면 저 내용에 대한 리뷰가 필요할 것 같아요.
사실상 제 현재 상황으로는 현재 코드를 리뷰하기에는 내용 이해가 어려워서...
혹시 https://gramian.de/ 에서 소개한 [원 논문](https://www.mdpi.com/1999-4893/11/7/91) 보신건가요?

Reply:
> 아하... 해당 사이트들은 논문과 부합하는건가요? 아니면 수정된 부분이 있나요?
> 
> 그리고 이 부분에 대한 설명도 부탁드립니다.
> 
> > 코드상으로 헷갈리는게, `ssp2` 의 아웃풋인 `y` 가 trajectory 인지, line 85의 덮어쓴 `y` 가 trajectory 인지도 헷갈리네요...

모두 다 논문과 부합하게 작성하였습니다.  `ssp2`의 아웃풋 `y`가 trajectory이고 line 85에서는 x의 범위 [-1, 1]로 각각을 나눕니다.

Reply:
> 

원 논문과 사이트에 나와있는 설명이 거의 동일하다고 보면 될 것 같습니다. 논문의 내용도 알고리즘 설명이 주제인 것 같아요. 이걸 먼저 같이 리뷰해볼까요 그럼?

Reply:
일단 이 코드는 거의 저 사이트에 나와있는 코드를 변환한 것과 같습니다. 괜찮다면 이 이슈를 머지하고 논문 리뷰 및 코드리뷰에 대한 이슈를 새로 파도 괜찮을까요?

Reply:
아하~
일단 우리 입장에서는 값만 제대로 나와도 충분(?)하긴 하니... ㅋㅋㅋㅋ
값이 제대로 나오는것도 확인되었고 하니, 리뷰하는걸 이슈로 파는 것 찬성합니다!

Reply:
거의 벡터화라는 코드를 고대로 가져와서 저렇게 바꿨다고 보시면 될 것 같아서 논문에 나온 코드와 달라진 점은 없을거에요 ㅎㅎ 리뷰를 새 이슈로 파겠습니다~